src/core/request.js

```diff
@@ -233,7 +233,9 @@ const request = async (ctx, uri, options) => {
       /* istanbul ignore else */
       if (opts.headers['transfer-encoding'] === undefined
         && opts.headers['content-length'] === undefined) {
-        opts.headers['content-length'] = Buffer.byteLength(opts.body, 'utf-8');
+        opts.headers['content-length'] = Buffer.isBuffer(opts.body)
+          ? opts.body.length
+          : Buffer.byteLength(opts.body, 'utf-8');
```

Comment:
for large binary buffers, this would create quite some overhead. and I'm not even sure, if the length will be correct, if the buffer contains illegal utf-8 chars. but I wasn't able to reproduce it.

Reply:
I agree, at this point `body` is either a buffer or a string and they should be handled differently, as suggested.